specs/vep-2420-getting-started-with-my-data/README.md

```diff
@@ -0,0 +1,198 @@
+    - [Poor documentation around secrets/properties](#Poor-documentation-around-secrets-and-properties)
+    - [Not trivial to run in the IDE](#Not-trivial-to-run-in-the-IDE)
+    - [Reliant on env variables](#Reliant-on-env-variables)
+  - [Requirements and goals](#requirements-and-goals)
+  - [High-level design](#high-level-design)
+  - [API design](#api-design)
+  - [Detailed design](#detailed-design)
+  - [Implementation stories](#implementation-stories)
+  - [Alternatives](#alternatives)
+
+## Summary
+
+Users struggle to get started with their own data in VDK.
+When evaluating a data platform like VDK one of the first actions is often to connect it to your own data sources and play around with it.
+Given that this is not a trivial task on VDK it most likely turns a lot of prospective users off.
+The issues range from:
+1. Poor assistance around configuration options
+2. Layers of abstraction aren't clear
+3. Poor documentation around secrets/properties
+4. Not trivial to RUN in IDE
+5. Reliant on env variables
+
+## Glossary
+
+## Motivation
+At the moment, VKD users face multiple challenges related to using their own data in VDK.
+Based on user feedback we've received, these can be split into 5 categories.
+
+### Poor assistance around configuration options
+
+VDK documents all config and all config can be seen with 'vdk config-help'.
+However despite this there are a number of issues which make it difficult to configure VDK.
+
+
+1. Users are expected to configure their job using a config.ini file. Ini files are not popular in the industry
+2. Sections are confusing
+ini files support sections (https://en.wikipedia.org/wiki/INI_file#Sections).
+Within VDK there are a few different sections, [vdk],[job], [owner].
+Newer properties always go in the [vdk] section.
+It's not clear which properties go in which section. In fact some properties can exist in different sections and will have the same result.
+3. There is no validation on the file at run time to make sure all properties exist. Ideally the system should throw an error if there are properties in the file which don't actually exist.
+4. There is no IDE assistance. Frameworks like spring support autocomplete in their property and yaml files.
+
+### Not obvious what properties do
+Below are the properties most jobs need to set.
+```bash
+export VDK_DB_DEFAULT_TYPE=SQLITE
+export VDK_INGEST_METHOD_DEFAULT=sqlite
+export VDK_INGEST_TARGET_DEFAULT=vdk-sqlite.db
+export VDK_SQLITE_FILE=vdk-sqlite.db
+```
+
+Ingest is a confusing term.
+It is a totally reasonable explanation to think that the ingestion database is the database where we want to ingest data from wheras it means the database we want to ingest data into.
+Its not obvious what impact having setting DB_DEFAULT_type has.
+
+### Layer's of abstraction are not clear
+
+The VDK advertises its self as being a useful tool for reading and writing to databases(It helps with retries etc...).
```

Comment:
Is the refering the the lack of Source concept in ingestion  which is on our roadmap https://github.com/vmware/versatile-data-kit/issues/2422. 

or something else?

Reply:
Yes having a source concept here would fix it.